hk-package-clean-except/action.yml

```diff
@@ -0,0 +1,98 @@
+name: "Clean all package versions except certain tags."
+description: "Action cleaning up all package versions published at ghcr.io except for certain tags."
+
+inputs:
+  package-name: 
+    description: "Package name."
+    required: true
+    type: string
+  token:
+    description: "Token with package deletion permissions."
+    required: true
+    type: string
+  tags-kept:
+    description: "Tags to be kept. Pass them as a list: e.g. 'latest, latest-unstable'."
+    required: true
+    type: string
+  package-org:
+    description: "Organization at which packages are published."
+    required: false
+    default: 'pyansys'
+    type: string
+  python-version:
+    description: 'Desired Python version.'
+    default: '3.10'
+    required: false
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+
+    - name: "Set up Python ${{ inputs.python-version }}"
+      uses: pyansys/actions/_setup-python@main
+      with:
+        python-version: ${{ inputs.python-version }}
+        use-cache: false
+
+    - name: "Install ghapi"
+      run: |
+        python -m pip install --upgrade pip
+        pip install ghapi
+
+    - name: "Clone script"
+      run: |
+        cat > clean-except.py << 'EOF'
+        
+        import os
+        
+        from ghapi.all import GhApi
+        from ghapi.core import print_summary
+        from ghapi.page import paged
+        
+        org_str = "${{ inputs.package-org }}"
+        pck_str = "${{ inputs.package-name }}"
+        valid_tags_str = "${{ inputs.tags-kept }}"

```

Comment:
One issue of the dochere strings is that you can not perform variable substitution.

A possible approach is to replace `${{ inputs.package-org }}` by `INPUTS_PACKAGE_ORG` and then run a `sed` command like:

```console
inputs_package_org=${{ inputs.package-org }}
sed -i "s|INPUTS_PACKAGE_ORG|$inputs_package_org|g" script.py
```

Previous approach does not allow direct GitHub substitution within the `sed` command. That is the reason of creating the `inputs_package_org` variable.

Reply:
Yeah, if that's the case I might go to env variables directly rather than running ``sed``...  Thanks for mentioning it @jorgepiloto